### PR TITLE
feat: get python dependencies from flow

### PR DESCRIPTION
**Pull request by HzaRashid**
Uses existing work on component-level package resolution to surface Python dependencies at the flow level and the current environment. Prioritizes installed distributions, followed by trusted overrides, and allows unsafe fall backs (using the package names from the flow json/dict). Adds unit tests to cover edge cases surrounding precedence and distribution resolution behavior.

## Summary by CodeRabbit

* **New Features**
  * Added improved package dependency resolution capabilities, including versioned distribution lookups and flow-based dependency computation with support for trusted versions and packages.

* **Tests**
  * Added comprehensive test coverage for dependency analysis, including edge cases for multiple versions, missing dependencies, and various resolution paths.

### CODE REVIEW

File: src/lfx/src/lfx/custom/utils.py [hunk 1]
```diff
@@ -874,4 +877,152 @@ def get_custom_component_template(component_cls):
...
+
+    If the flow specifies multiple
+    versions of the same package,
+    it will return all the versions
+    in no particular order.
+    This may cause non-deterministic
+    behavior downstream, e.g., building
+    a requirements.txt file.
+    The caller is responsible
+    for deduplicating versions.
```
Comment (jordanrfrazier) [inline src/lfx/src/lfx/custom/utils.py:925]: I like how the docs are written like a poem with multiple stanzas

File: src/lfx/tests/unit/custom/test_utils_metadata.py
Comment (ogabrielluiz) [inline src/lfx/tests/unit/custom/test_utils_metadata.py:1]: We should probably reduce the amount of mocking here in these tests. Most of the data we need to validate can be accessed at the time of testing.
